src/types/input.jl

```diff
@@ -2,81 +2,151 @@
 
 abstract type InputType end
 
-struct Bosonic <:InputType
-end
-struct PartDist <:InputType
-end
-abstract type ToyModel <: InputType
-end
-abstract type RandomModel <: InputType
-end
-struct Distinguishable <:InputType
+abstract type Bosonic <: InputType end
```

Comment:
shouldn't be an abstract type, see comment below

Reply:
If this is not the case then we cannot state that ToyModel or RandomModel are subtypes of ParDist and in this way I do not understand the point of defining ParDist if we define them as subtypes of InputType